pkg/metrics/statsd/cosmosdb/metrics.go

```diff
@@ -0,0 +1,77 @@
+package cosmodb
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Azure/ARO-RP/pkg/metrics"
+)
+
+var actions = map[string]bool{
+	"colls":    true,
+	"docs":     true,
+	"triggers": true,
+	"dbs":      true,
+}
+
+var tables = map[string]bool{
+	"AsynOperations":    true,
+	"OpenShiftClusters": true,
+	"Subscriptions":     true,
+}
+
+var _ http.RoundTripper = (*tracerRoundTripper)(nil)
+
+type tracerRoundTripper struct {
+	m  metrics.Interface
+	tr http.RoundTripper
+}
+
+func New(tr *http.Transport, m metrics.Interface) *tracerRoundTripper {
+	return &tracerRoundTripper{
+		m:  m,
+		tr: tr,
+	}
+}
+
+func (t *tracerRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
+	start := time.Now()
+	resp, err := t.tr.RoundTrip(r)
+
+	action := "unknown"
+	table := "unknown"
+	code := "unknown"
+
+	match := strings.Split(r.URL.String(), "/")
```

Comment:
as a first cut tempted to simplify and follow the k8s path approach - just remove the last uid as discussed on slack to reduce cardinality as a starting point